src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Simplification/Simplifiers/CastSimplifier.cs

```diff
@@ -497,10 +505,19 @@ private static bool IsInTargetTypingLocation(ExpressionSyntax node)
             return false;
         }
 
-        private static bool IsExplicitCastThatMustBePreserved(Conversion conversion)
+        private static bool IsExplicitCastThatMustBePreserved(ExpressionSyntax castNode, Conversion conversion)
         {
             if (conversion.IsExplicit)
             {
+                // If it's an explicit nullable conversion in a conditional expression, then this may still be removable if it's C# 9,
+                // and there's target typing involved. We'll have to go through the fancier logic in this case.
```

Comment:
so i really like when an example is given to ground exactly what your'e taking about.